plugins/jobs/src/js/data/JobModel.ts

```diff
@@ -76,17 +80,119 @@ function isJobQueryArg(arg: any): arg is JobQueryArgs {
   return arg.id !== undefined;
 }
 
+// Sort and filtering
+function sortJobs(
+  jobs: Job[],
+  sortBy: SortOption = "ID",
+  sortDirection: SortDirection = "ASC"
+): Job[] {
+  jobs.sort((a, b) => {
+    let result;
+
+    switch (sortBy) {
+      case "ID":
+        result = sortJobById(a, b);
+        break;
+      case "STATUS":
+        result = sortJobByStatus(a, b);
+        break;
+      case "LAST_RUN":
+        result = sortJobByLastRun(a, b);
+        break;
+      default:
+        result = 0;
+    }
+
+    const direction = sortDirection === "ASC" ? 1 : -1;
+    return result * direction;
+  });
+
+  return jobs;
+}
+
+function sortJobById(a: Job, b: Job): number {
+  return a.id.localeCompare(b.id);
+}
+
+function sortJobByStatus(a: Job, b: Job): number {
+  return (
+    JobStates[a.scheduleStatus].sortOrder -
+    JobStates[b.scheduleStatus].sortOrder
+  );
+}
+
+function sortJobByLastRun(a: Job, b: Job): number {
+  return (
+    JobStatus[a.lastRunStatus.status].sortOrder -
+    JobStatus[b.lastRunStatus.status].sortOrder
+  );
+}
+
+function filterJobsById(
+  jobs: MetronomeJobResponse[],
+  filter: string | null
+): MetronomeJobResponse[] {
+  if (filter === null) {
+    return jobs;
+  }
+
+  return jobs.filter(({ id }) => id.indexOf(filter) !== -1);
+}
+
+function filterJobsByNamespace(
+  jobs: MetronomeJobResponse[],
+  namespace: string | null
+): MetronomeJobResponse[] {
+  if (!namespace) {
+    return jobs;
+  }
+
+  return jobs.filter(job => {
+    return job.id.startsWith(namespace);
+  });
+}
+
 export const resolvers = ({
+  fetchJobs,
   fetchJobDetail,
   pollingInterval
 }: ResolverArgs): IResolvers => ({
   Query: {
     jobs(
-      _obj = {},
-      _args: GeneralArgs = {},
+      _parent = {},
+      args: GeneralArgs = {},
       _context = {}
     ): Observable<JobConnection | null> {
-      return Observable.of(null);
+      const {
+        sortBy = "ID",
+        sortDirection = "ASC",
+        filter = null,
+        namespace = null
+      } = args as JobsQueryArgs;
+
+      const pollingInterval$ = Observable.timer(0, pollingInterval);
+      const responses$ = pollingInterval$.exhaustMap(fetchJobs);
+
+      const jobsInNamespace$ = responses$.map(jobs =>
+        filterJobsByNamespace(jobs, namespace)
+      );
+      const count$ = jobsInNamespace$.map(jobs => jobs.length);
+
+      return (
+        jobsInNamespace$
+          .map(jobs => filterJobsById(jobs, filter))
+          //TODO We need to remove the N + 1 query
+          // https://jira.mesosphere.com/browse/DCOS-38201
+          .map(jobs => jobs.map(job => fetchJobDetail(job.id)))
+          .switchMap(obs => Observable.combineLatest(obs))
```

Comment:
I'm sorry but when I see `jobs` and `obs` close to each other I can't stop thinking there's a `j` missing. Let's not abbreviate variables 💪 plus if argument is an Observable we may wanna consider adding `$` at the end